<?xml version="1.0" encoding="UTF-8"?>

<launch>

    <node name="map_to_base_link"   pkg="tf2_ros"       type="static_transform_publisher"  	args="1.0 0.0 0.0 0 0 0 map base_link" />
	
	<!--<node pkg="tf2_ros" type="static_transform_publisher" name="base_to_laser_pub" args="0.2 0.0 -0.05 3.141592654 0 0 base_link lidar_link" />-->
	
	<node name="base_to_laser_pub"  pkg="tf2_ros"       type="static_transform_publisher"   args="0.2 0.0 -0.05 0 0 3.141592654 base_link lidar_link" />
	
	<node name="ydlidar_node"       pkg="ydlidar_ros"   type="ydlidar_node"                 output="screen" respawn="false" >
        
        <param name="port"              type="string"   	value="/dev/ttyUSB0"/>  
        <param name="baudrate"          type="int"      	value="230400"/>  
        <param name="frame_id"          type="string"   	value="lidar_link"/>
        <param name="resolution_fixed"  type="bool"     	value="true"/>
        <param name="auto_reconnect"    type="bool"     	value="true"/>
        <param name="reversion"         type="bool"     	value="true"/>
        <param name="angle_min"         type="double"   	value="-90" />
        <param name="angle_max"         type="double"   	value="90" />
        <param name="range_min"         type="double"   	value="0.1" />
        <param name="range_max"         type="double"   	value="10.0" />
        <param name="ignore_array"      type="string"   	value="" />
        <param name="frequency"         type="double"   	value="5"/>
        <param name="samp_rate"         type="int"      	value="4"/>
    
    </node>
	
	<node name="lidar_supervisor"	pkg="lidar_light"	type="lidar_supervisor"				args=""/>

</launch>
